File: c/resistor-color/src/resistor_color.c
#include "resistor_color.h"

static resistor_band_t all_colors[] =
{
    BLACK,
    BROWN,
    RED,
    ORANGE,
    YELLOW,
    GREEN,
    BLUE,
    VIOLET,
    GREY,
    WHITE
};

int color_code(resistor_band_t color)
{
    return color;
}

resistor_band_t* colors()
{
    return all_colors;
}

File: dart/resistor-color/lib/resistor_color.dart
/// Responsible for parsing the resistor's colors.
class ResistorColor {
  /// Complete resistor's color code.
  final colors = const [
    'black',
    'brown',
    'red',
    'orange',
    'yellow',
    'green',
    'blue',
    'violet',
    'grey',
    'white'
  ];

  /// Returns the color code for a given [color].
  int colorCode(String color) => colors.indexOf(color);
}
